### PR TITLE
Return when removePod failed

### PR DESCRIPTION
**Pull request by cwdsuzhou**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>

/kind bug 

**What this PR does / why we need it**:

Return when removePod failed

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by alculquicondor** [comment]
Have you encountered a path that triggers this error?
It looks to my like it shouldn't.
If so, please add a test case. If not, please change this to kind cleanup

**Comment by cwdsuzhou** [comment]
> Have you encountered a path that triggers this error?
> It looks to my like it shouldn't.
> If so, please add a test case. If not, please change this to kind cleanup

Thanks, have changed the kind

**Comment by ahg-g** [comment]
I don't think this is necessary, the code should try to add the pod even if the remove fails.

**Comment by alculquicondor** [comment]
As I said, I don't think this codepath is ever hit. But that doesn't mean we should ignore the error, because something might change in the implementation and then we would be obscuring the error. If we return it, we have the chance to catch it during unit tests.
Otherwise, let's log it?

**Comment by cwdsuzhou** [comment]
ping @ahg-g @k82cn  for approval

File: pkg/scheduler/internal/cache/cache.go [hunk 1]
```diff
@@ -407,7 +407,9 @@ func (cache *schedulerCache) AddPod(pod *v1.Pod) error {
 			// The pod was added to a different node than it was assumed to.
 			klog.Warningf("Pod %v was assumed to be on %v but got added to %v", key, pod.Spec.NodeName, currState.pod.Spec.NodeName)
 			// Clean this up.
-			cache.removePod(currState.pod)
+			if err = cache.removePod(currState.pod); err != nil {
+				return err
```
Comment (ahg-g) [inline pkg/scheduler/internal/cache/cache.go]: change this to a ```klog.Errorf("removing pod: %v", err)``` instead of returning an error.

Comment (cwdsuzhou) [reply pkg/scheduler/internal/cache/cache.go]: done, thanks!

Comment (cwdsuzhou) [reply pkg/scheduler/internal/cache/cache.go]: ping @ahg-g for review

**Pull request merged**